pkg/populator/populator.go

```diff
@@ -85,6 +87,11 @@ func (p *Populator) handlePVUpdate(pv *v1.PersistentVolume) {
 			if !found {
 				return
 			}
+			if p.UseNodeNameOnly && strings.HasPrefix(provisioner, p.Name) {
+				// The PV was created on this node
+				p.Cache.AddPV(pv)
+				return
+			}
```

Comment:
I'm afraid we cann't rely on this, as one node name can be the prefix of another node name.

Reply:
This works only if `config.NodeLabelsForPV` is enabled, otherwise `kubernetes.io/hostname` label will not be added to PV object. However, it's a safety check.